remote/remote_state.go

```diff
@@ -79,28 +81,46 @@ func (remoteState *RemoteState) NeedsInit(terragruntOptions *options.TerragruntO
 	if initializer, hasInitializer := remoteStateInitializers[remoteState.Backend]; hasInitializer {
 		// Remote state initializer says initialization is necessary
 		return initializer.NeedsInitialization(remoteState.Config, state.Backend, terragruntOptions)
-	} else if state.IsRemote() && remoteState.differsFrom(state.Backend, terragruntOptions) {
+	} else if state.IsRemote() {
+		needsInit, err := remoteState.differsFrom(state.Backend, terragruntOptions)
+		if nil != err {
+			return false, err
+		}
 		// If there's no remote state initializer, then just compare the the config values
-		return true, nil
+		return needsInit, nil
```

Comment:
Is this change related to this PR?

Reply:
Yes it definitly relates to this PR because I had to extend the return value of `remote_state.go::NeedsInit` with an `error` object because the method must now do a late interpolation https://github.com/tmeckel/terragrunt/blob/8066792ee5909fee702b6e03ee4a88e2cb56ca2d/remote/remote_state.go#L71 and the interpolation might result in an error. In addtion based on the interpolation of the settings for the remote state implementation it might need an initialization or not.